std/file.d

```diff
@@ -456,7 +456,7 @@ Params:
 Throws: $(D FileException) on error.
  */
 void append(R)(R name, const void[] buffer)
-    if (isInputRange!R && isSomeChar!(ElementEncodingType!R) || isSomeString!R)
+    if (_isNonAliasedCharInputRange!R)
```

Comment:
> I didn't want to do this b/c it will show up in the documentation.

Yeah, I didn't like that aspect. too. A verbose alternative would be to put a version with a readable contraint into a version(StdDdoc) condition.